controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java

```diff
@@ -458,6 +461,294 @@ public void retentionStreamTest() throws Exception {
         assertTrue(truncProp.getProperty().getStreamCut().get(0) == 1L && truncProp.getProperty().getStreamCut().get(1) == 1L);
     }
 
+    @Test(timeout = 30000)
+    public void sizeBasedRetentionStreamTest() throws Exception {
+        final ScalingPolicy policy = ScalingPolicy.fixed(2);
+        final RetentionPolicy retentionPolicy = RetentionPolicy.builder()
+                .retentionType(RetentionPolicy.RetentionType.SIZE).retentionParam(100L).build();
+
+        String streamName = "test";
+        final StreamConfiguration configuration = StreamConfiguration.builder().scope(SCOPE).streamName(streamName).scalingPolicy(policy)
+                .retentionPolicy(retentionPolicy).build();
+
+        streamStorePartialMock.createStream(SCOPE, streamName, configuration, System.currentTimeMillis(), null, executor).get();
+        streamStorePartialMock.setState(SCOPE, streamName, State.ACTIVE, null, executor).get();
+
+        assertNotEquals(0, consumer.getCurrentSegments(SCOPE, streamName).get().size());
+        WriterMock requestEventWriter = new WriterMock(streamMetadataTasks, executor);
+        streamMetadataTasks.setRequestEventWriter(requestEventWriter);
+
+        // region size based retention on stream cuts on epoch 0
+        // region no previous streamcut
+        // first retention iteration
+        // streamcut1: 19 bytes(0/9,1/10)
+        long recordingTime1 = System.currentTimeMillis();
+        Map<Integer, Long> map1 = new HashMap<>();
+        map1.put(0, 9L);
+        map1.put(1, 10L);
+        long size = streamStorePartialMock.getSizeTillStreamCut(SCOPE, streamName, map1, null, executor).join();
+        assertEquals(size, 19);
+        StreamCutRecord streamCut1 = new StreamCutRecord(recordingTime1, size, map1);
+
+        doReturn(CompletableFuture.completedFuture(streamCut1)).when(streamMetadataTasks).generateStreamCut(
+                anyString(), anyString(), any());
+
+        streamMetadataTasks.retention(SCOPE, streamName, retentionPolicy, recordingTime1, null).get();
+        // verify that one streamCut is generated and added.
+
```

Comment:
I think the `// verify...` comment refers to lines 499-500. This new line is confusing.